### PR TITLE
Enable test cases that were disabled, add some test cases

### PR DESCRIPTION
**Pull request by Hibeom0929**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Role.java [hunk 1]
```diff
@@ -12,7 +12,7 @@ public enum Role {
     PROFESSOR;
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Roles should be either 'STUDENT', 'TA', or 'PROFESSOR'.";
+            "Roles should be either 'STUDENT', 'TA', or 'PROFESSOR', or an unambiguous prefix of it.";;
```
Comment (E0735389) [inline src/main/java/seedu/address/model/person/Role.java:15]: There's a reason why both messages exists.

Previously if the JSON contains something like `s` instead of `STUDENT` as the role, the error message being thrown would be `Roles should be either 'STUDENT', 'TA', or 'PROFESSOR'.`. With this change, the error message would be incorrect.

Comment (Hibeom0929) [reply src/main/java/seedu/address/model/person/Role.java:15]: Oh that's why there was a separate error message for the role in ParseUtil. I'll fix it

File: src/main/java/seedu/address/model/person/Role.java [hunk 2]
```diff
@@ -12,7 +12,7 @@ public enum Role {
     PROFESSOR;
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Roles should be either 'STUDENT', 'TA', or 'PROFESSOR'.";
+            "Roles should be either 'STUDENT', 'TA', or 'PROFESSOR'.";;
```
Comment (LargeCrowd) [inline src/main/java/seedu/address/model/person/Role.java:15]: why did you add another `;`?

Comment (Hibeom0929) [reply src/main/java/seedu/address/model/person/Role.java]: Typo. Fixed

**Review by E0735389 — APPROVED** [review]
